lib/token.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const BaseModel = require('./base');
+const nodeify = require('./nodeify');
+const iron = require('iron');
+
+// Symbols for private members
+const password = Symbol('password');
+
+class TokenModel extends BaseModel {
+    /**
+     * Construct a TokenModel object
+     * @method constructor
+     * @param  {Object}    config
+     * @param  {Object}    config.datastore         Object that will perform operations on the datastore
+     * @param  {Number}    config.userId            The ID of the associated user
+     * @param  {String}    config.value             The token value
+     * @param  {String}    config.tokenId           The token ID
+     * @param  {String}    config.name              The token name
+     * @param  {String}    config.description       The token description
+     * @param  {Date}      config.lastUsed          The last time the token was accessed
+     * @param  {String}    config.password          The encryption password
+     */
+    constructor(config) {
+        super('token', config);
+        this[password] = config.password;
+    }
+
+    /**
+     * Update a token with sealed token value
+     * @method update
+     * @return {Promise}
+     */
+    update() {
+        return nodeify.withContext(iron, 'seal', [this.value, this[password], iron.defaults])
```

Comment:
For changing: I was talking to Dao and Min yesterday, and they pointed out that github has an option to regenerate a token, which changes the value while leaving the name/description intact. If we wanted to add that, that would do it, but I might be getting ahead of myself there.

For keeping only the sealed version in value, I was following the pattern secrets use, which seems to be that when you retrieve it from the datastore you unseal it so that you can actually use the value, and then you seal it again when the model is put back into the datastore.